pkg/oceancd/model/rollout/rollout.go

```diff
@@ -0,0 +1,133 @@
+	ManualPausing        Status = "manualPausing"
+	InvalidSpec          Status = "invalidSpec"
+	Deallocating         Status = "deallocating"
+	Verifying            Status = "verifying"
+	FailurePolicyPaused  Status = "failurePolicyPaused"
+	FailurePolicyPausing Status = "failurePolicyPausing"
+	BackgroundVerifying  Status = "backgroundVerifying"
+)
+
+const (
+	PhaseStatusFinished     = "finished"
+	PhaseStatusPromoted     = "promoted"
+	PhaseStatusFullPromoted = "fullPromoted"
+	PhaseStatusPromoting    = "promoting"
+	PhaseStatusPaused       = "paused"
+	PhaseStatusInProgress   = "inProgress"
+	PhaseStatusAborted      = "aborted"
+	PhaseStatusCanceled     = "canceled"
+	PhaseStatusPending      = "pending"
+	PhaseStatusDropped      = "dropped"
+	PhaseStatusVerifying    = "verifying"
+	PhaseStatusAborting     = "aborting"
+
+	VerificationStatusRunning    = "running"
+	VerificationStatusSuccessful = "successful"
+	VerificationStatusFailed     = "failed"
+	VerificationStatusError      = "error"
+)
+
+var UncompletedPhaseStatuses = []string{PhaseStatusPromoting, PhaseStatusPaused, PhaseStatusInProgress, PhaseStatusPending,
+	PhaseStatusVerifying, PhaseStatusAborting}
+var CompletedStatuses = []Status{Failed, Aborted, Finished, Canceled}
+var VerificationStatusOrder = map[string]int{
+	VerificationStatusFailed:     1,
+	VerificationStatusError:      2,
+	VerificationStatusRunning:    3,
+	VerificationStatusSuccessful: 4}
+
+type Status string
+
+func (s Status) IsCompleted() bool {
+	for _, completedStatus := range CompletedStatuses {
+		if s == completedStatus {
+			return true
+		}
+	}
+
+	return false
+}
+
+type ReplicasInfo struct {
+	Desired    int `json:"desired"`
+	Ready      int `json:"ready"`
+	InProgress int `json:"inProgress"`
+	Failed     int `json:"failed"`
+}
+
+type VersionStatus struct {
+	Version           string       `json:"version"`
+	K8sService        string       `json:"k8sService"`
+	TrafficPercentage int          `json:"trafficPercentage"`
+	Replicas          ReplicasInfo `json:"replicas"`
+}
+
+type Rollout struct {
+	ID                        string        `json:"id"`
+	Status                    Status        `json:"status"`
+	SpotDeployment            string        `json:"spotDeployment"`
+	OriginalRolloutId         string        `json:"originalRolloutId"`
+	NewRolloutId              string        `json:"newRolloutId"`
+	StartTime                 string        `json:"startTime"`
+	EndTime                   string        `json:"endTime"`
+	ClusterId                 string        `json:"clusterId"`
+	Namespace                 string        `json:"namespace"`
+	Strategy                  string        `json:"strategy"`
+	HasBackgroundVerification bool          `json:"hasBackgroundVerification"`
+	NewVersionStatus          VersionStatus `json:"newVersionStatus"`
+	StableVersionStatus       VersionStatus `json:"stableVersionStatus"`
+	Phases                    []Phase       `json:"phases"`
```

Comment:
This is not part of Rollout object
I think you should create a wrapper called DetailedRollout that includes Rollout and Phase list
wdyt?

Reply:
I think so